src/journal/journald-context.c

```diff
@@ -784,3 +840,40 @@ void client_context_acquire_default(Server *s) {
 
         }
 }
+
+#if HAVE_PCRE2
+static bool is_log_matched(pcre2_code *regex, const char *message) {
+        int r;
+        _cleanup_(sym_pcre2_match_data_freep) pcre2_match_data *md = NULL;
+
+        assert(regex);
+
+        md = sym_pcre2_match_data_create(1, NULL);
+        if (!md)
+                return log_oom();
+
+        r = sym_pcre2_match(regex, (PCRE2_SPTR8) message, strlen(message), 0, 0, md, NULL);
+        if (r == PCRE2_ERROR_NOMATCH) {
+                return false;
+        } else if (r < 0) {
+                log_warning_errno(r, "Failed to match log message with log regex: %m");
+                return false;
+        }
+
+        return true;
+}
+#endif
+
+bool is_log_discarded(ClientContext *c, const char *message) {
+#if HAVE_PCRE2
+        assert(c);
+
+        if (c->log_include_regex && !is_log_matched(c->log_include_regex, message))
+                return true;
```

Comment:
Should check here for is_log_matched() > 0 because it can return a negative errno value as well